packages/shared/components/MainMenu.svelte

```diff
@@ -1,78 +1,75 @@
 <script lang="typescript">
-    import { Drawer, Icon, Text } from 'shared/components'
-    import { logout } from 'shared/lib/app'
-    import { isBright, getInitials } from 'shared/lib/helpers'
+    import { Drawer, Icon, Text, ProfileActions } from 'shared/components'
+    import { getInitials } from 'shared/lib/helpers'
     import { activeProfile, getColor } from 'shared/lib/profile'
     import { selectedAccount } from 'shared/lib/wallet'
     import { Settings } from 'shared/routes'
-    import { settingsRoute, settingsRouter, SettingsRoute } from '@core/router'
-    import { Locale } from '@core/i18n'
+    import {
+        profileRoute,
+        profileRouter,
+        ProfileRoute,
+        settingsRoute,
+        settingsRouter,
+        SettingsRoute,
+    } from '@core/router'
+    import { localize } from '@core/i18n'
 
-    export let locale: Locale
-
-    $: color = getColor($activeProfile, $selectedAccount?.id) as string
-    $: textColor = isBright(color) ? 'gray-800' : 'white'
+    $: profileColor = getColor($activeProfile, $selectedAccount?.id) as string
 
     let drawer: Drawer
 
     $: profileInitial = getInitials($activeProfile?.name, 1)
 
     function handleBackClick() {
-        if ($settingsRoute === SettingsRoute.Init) {
-            drawer?.close()
+        if ($profileRoute === ProfileRoute.ProfileActions) {
+            drawer.close()
+        } else if ($settingsRoute === SettingsRoute.Init) {
+            $profileRouter.previous()
         } else {
             $settingsRouter.goTo(SettingsRoute.Init)
         }
     }
 
     function handleClick() {
-        $settingsRouter.goTo(SettingsRoute.Init)
+        $profileRouter.goTo(ProfileRoute.ProfileActions)
         drawer.open()
     }
+
+    function handleSettingsClick() {
```

Comment:
return types

Reply:
Added, but I think we could avoid to type `void` since the type is also introspected by Typescript without adding it.